python/lsst/ts/mtrotator/rotator_csc.py

```diff
@@ -215,13 +222,39 @@ async def check_ccw_following_error(self) -> None:
         except Exception:
             self.log.exception("check_ccw_following_error failed")
 
+    async def detect_vibration(self) -> None:
+        """Detect the vibration."""
+
+        assert self._vibration_detector is not None
+
+        frequencies = self._vibration_detector.check_vibration_frequency()
+
+        for frequency in frequencies:
+            # This is to keep the backward compatibility with ts_xml v22.0.0
+            # that does not have the 'lowFrequencyVibration' defined in xml.
+            if hasattr(self, "evt_lowFrequencyVibration"):
+                await self.evt_lowFrequencyVibration.set_write(frequency=frequency)
```

Comment:
Just one think to keep in mind here... The way this is written down it is going to cause a burst of events being published when there is more than one detected frequency. This is fine control-system wise but keep in mind that the EFD has a time window of some milliseconds (don't remember the exact value), which means there's a high likelihood the some events will be discarded if you publish them too fast. 

I recommend you add an `await asyncio.sleep(0.1)` after every call here to make sure you give it enough time between the data.

Reply:
Improved in the commit: [ba553ac](https://github.com/lsst-ts/ts_mtrotator/pull/92/commits/ba553ac1e54428b0af284a65311b88c4367b4361). Thanks!